### ISSUE TITLE
New lint: function takes slice and treats it as an array

### ISSUE THREAD

**Issue by paolobarbolini** [rust-lang/rust-clippy#8397]
### What it does

Finds functions expecting a contiguous view of items of a `const` length taking a `slice` instead of an `array`.

Real world case: https://github.com/image-rs/jpeg-decoder/pull/215/files

### Lint Name

fn_slice_param_is_array

### Category

pedantic

### Advantage

- Removes bounds checks/panics by giving the caller the responsibility to give the correct size
- Removes potential programming mistakes

### Drawbacks

- Possible false positives?

### Example

```rust
const LEN: usize = 64;

fn foo(items: &[u8]) {
    assert_eq!(items.len(), LEN);
    // or...
    debug_assert_eq!(items.len(), LEN);
    // or...
    let items = &items[..LEN];
}
```

Could be written as:

```rust
const LEN: usize = 64;

fn foo(items: &[u8; LEN]) {
}
```
